rest_framework/filters.py

```diff
@@ -156,26 +156,36 @@ def get_default_ordering(self, view):
         return ordering
 
     def remove_invalid_fields(self, queryset, fields, view):
-        valid_fields = getattr(view, 'ordering_fields', self.ordering_fields)
+        ordering_fields = getattr(view, 'ordering_fields', self.ordering_fields)
 
-        if valid_fields is None:
-            # Default to allowing filtering on serializer fields
-            serializer_class = getattr(view, 'serializer_class')
+        if not ordering_fields == '__all__':
+            serializer_class = getattr(view, 'serializer_class') or view.get_serializer_class()
```

Comment:
`getattr` will raise an exception (an `AttributeError`, usually) when the requested attribute is not found on the object. in this case, you are probably looking for something like

```python
serializer_class = view.get_serializer_class() if hasattr(view, 'get_serializer_class) else view.serializer_class
```

or the rough equivalent of

```python
serializer_class = getattr(view, 'serializer_class', None)

if serializer_class is None:
    serializer_class = view.get_serializer_class()
```

Then again, it may just be better to prefer `view.get_serializer_class()` instead of relying on the `serializer_class` attribute.